CI/docker/Dockerfile.gpu-inference

```diff
@@ -1,4 +1,4 @@
-FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.0.1-gpu-py310-cu118-ubuntu20.04-sagemaker
+FROM 763104351884.dkr.ecr.us-east-1.amazonaws.com/pytorch-inference:2.1.0-gpu-py310-cu118-ubuntu20.04-sagemaker
```

Comment:
Why don't we use the same cuda version for training and inference?

Reply:
Currently the base DLC inference images that are available only have support for CUDA 11.8
ref:
https://github.com/aws/deep-learning-containers/blob/master/available_images.md#ec2-framework-containers-tested-on-ec2-ecs-and-eks-only

Reply:
Let's make sure the base image CUDA and torch version are consistent. Otherwise, we might have gaps and issues during hosting

Reply:
Interesting, then that would mean `tensorrt` would fail with the segmentation fault on the inference image.